quantum/rgb_matrix_animations/typing_heatmap_anim.h

```diff
@@ -38,6 +43,10 @@ bool TYPING_HEATMAP(effect_params_t* params) {
         memset(g_rgb_frame_buffer, 0, sizeof g_rgb_frame_buffer);
     }
 
+    // Increment the tick count, it's ok to overflow since we're just using mod
+    // to decrease the heatmap values every N ticks.
+    tick++;
```

Comment:
This function is run multiple times to render a single rgb effect frame. Move this tick counter up into the `if (params->init)`, additionally, compute and store the result from `(tick % RGB_MATRIX_TYPING_HEATMAP_DECREASE_DELAY == 0)` ahead of time instead of doing that in the loop.

Reply:
Moving it inside the `if` breaks the code. Pretty sure `init` is only set to true whenever the RGB effect starts for the first time.